### PR TITLE
resource_control: support set resource limit for background tasks

### LINKED ISSUE #56019: resource_control: Add resource limit for background task
## Feature Request

**Is your feature request related to a problem? Please describe:**

Current, the background task control of resource_control is based on the total resource quota(CPU, IO) of each tikv instance. But in real use cases, foreground tail lantency increase significantly before resource is fully used. So we need to set some kind of limit that background tasks can use so user can control the performance impact by background tasks.

Current background settings:
```
ALTERR RESOURCE GROUP default BACKGROUND=(TASK_TYPES="br,import");
``` 
New SQL pattern:
```
ALTERR RESOURCE GROUP default BACKGROUND=(TASK_TYPES="br,import", UTILIZATION_LIMIT=30);
```
We add a new field `UTILIZATION_LIMIT` (for the `default` resource group) to control the resource percentage that background tasks can use. 

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #56019

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by glorv** [comment]
@nolouch @Connor1996 @HuSharp PTAL, thanks

File: pkg/domain/infosync/info.go [hunk 1]
```diff
@@ -623,6 +623,7 @@ func ModifyResourceGroup(ctx context.Context, group *rmpb.ResourceGroup) error {
 	if err != nil {
 		return err
 	}
+	logutil.BgLogger().Info("ModifyResourceGroup", zap.Stringer("group", group))
```
[the remark this answers is not in the document]
Comment (glorv) [reply pkg/domain/infosync/info.go:626]: sorry, redundant log for debug, I'll remove it.

**Comment by glorv** [comment]
@nolouch @JmPotato  PTAL again

**Comment by glorv** [comment]
@nolouch PTAL again, thanks~

File: pkg/parser/parser_test.go [hunk 2]
```diff
@@ -3873,9 +3873,13 @@ func TestDDL(t *testing.T) {
...
 		{`create resource group x ru_per_sec=1000 QUERY_LIMIT (EXEC_ELAPSED "10s" ACTION COOLDOWN WATCH EXACT DURATION='10m')  background (task_types 'br,lightning')`, true, "CREATE RESOURCE GROUP `x` RU_PER_SEC = 1000, QUERY_LIMIT = (EXEC_ELAPSED = '10s' ACTION = COOLDOWN WATCH = EXACT DURATION = '10m'), BACKGROUND = (TASK_TYPES = 'br,lightning')"},
 		{`create resource group x ru_per_sec=1000 QUERY_LIMIT (EXEC_ELAPSED "10s" ACTION COOLDOWN WATCH PLAN DURATION='10m')  background (task_types 'br,lightning')`, true, "CREATE RESOURCE GROUP `x` RU_PER_SEC = 1000, QUERY_LIMIT = (EXEC_ELAPSED = '10s' ACTION = COOLDOWN WATCH = PLAN DURATION = '10m'), BACKGROUND = (TASK_TYPES = 'br,lightning')"},
+		{`create resource group x ru_per_sec=1000 QUERY_LIMIT (EXEC_ELAPSED "10s" ACTION COOLDOWN WATCH PLAN DURATION='10m')  background (task_types 'br,lightning', utilization_limit 10)`, true, "CREATE RESOURCE GROUP `x` RU_PER_SEC = 1000, QUERY_LIMIT = (EXEC_ELAPSED = '10s' ACTION = COOLDOWN WATCH = PLAN DURATION = '10m'), BACKGROUND = (TASK_TYPES = 'br,lightning', UTILIZATION_LIMIT = 10)"},
```
Comment (nolouch) [inline pkg/parser/parser_test.go:3882]: will it report `unsupported operation. Currently, only the default resource group support change background setting` under the parser layer?

**Review by nolouch — APPROVED** [review]

File: pkg/parser/parser_test.go (hunk 2 above)
Comment (glorv) [reply pkg/parser/parser_test.go:3882]: Its format is valid so it can be parsed, `unsupported operation` is report at ddl module.

**Comment by glorv** [comment]
@tangenta  PTAL, thanks

**Review by tangenta — APPROVED** [review]

**Comment by glorv** [comment]
@yudongusa PTAL

**Review by yudongusa — APPROVED** [review]
LGTM

**Pull request merged**